src/api.c

```diff
@@ -2755,6 +2795,10 @@ static int cgroup_find_parent(struct cgroup *cgroup, char *controller,
 	cgroup_dbg("parent's name is %s\n", parent_path);
 
 	if (stat(child_path, &stat_child) < 0) {
+		if (is_cgrp_ctrl_mnt_shared(cgroup, controller)) {
```

Comment:
More thinking out loud here... What if we kept track of all of the deleted paths?  Then we could compare the `child_path` with previously deleted paths.  If it's already been deleted, we can probably just `continue` and move on.

Thoughts?

Reply:
I did think about it and was biased towards the current implementation, due to the frequency of such use cases.  The check gets performed on error,  to confirm if the directory was deleted by the previous controller or if it's a real bug.  Another approach I was thinking of is adding `cg_mount_point->shared` flag, which gets set during `cgroup_init` and we would quickly check if it's a shared mount point, again in this approach we are assuming that a first of the controller sharing the mount point deleted it.

Reply:
The `shared` flag is an interesting idea

Reply:
I have implemented the `shared_mnt` flag idea at https://github.com/kamalesh-babulal/libcgroup/commit/6e156aa1c1e195a60e811e1f4a541620356d76a1 please let me know if you like it better than the current approach.